### PR TITLE
Allow to provide an ACL flag

### PR DESCRIPTION
**Pull request by cweinberger**

### CODE REVIEW

File: Sources/Storage/NetworkDriver.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@ import Foundation
 public protocol NetworkDriver: Service {
     var pathBuilder: PathBuilder { get set }
 
-    func upload(entity: inout FileEntity, on container: Container) throws -> Future<String>
+    func upload(entity: inout FileEntity, access: AccessControlList, on container: Container) throws -> Future<String>
```
Comment (siemensikkema) [inline Sources/Storage/NetworkDriver.swift:8]: this is a breaking change, are you planning to bump the major version? Do any of our other packages depend on it?

Comment (cweinberger) [reply Sources/Storage/NetworkDriver.swift:8]: yes, just let us do a major version bump to be safe. It only applies if you dont use Storage but the driver directly

**Comment by siemensikkema** [comment]
@cweinberger the annoying thing with this package is that it tries to be cloud-agnostic but in reality it's just for AWS. The AccessControlList is an AWS specific type, right?

File: Sources/Storage/NetworkDriver.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/Storage/NetworkDriver.swift:8]: exactly and no-one is probably doing that

**Comment by cweinberger** [comment]
> @cweinberger the annoying thing with this package is that it tries to be cloud-agnostic but in reality it's just for AWS. The AccessControlList is an AWS specific type, right?

Yes it's AWS specific. Never came to my mind as we only use it ever with AWS. What do we do?

**Review by siemensikkema — APPROVED** [review]
Let's just move on with this and create a major version. It's not worth trying to be too puristic for this older package 🙂

**Pull request merged**